src/controllers/components/model.js

```diff
@@ -1,12 +1,30 @@
-import { searchModel } from '../../lib/api-client';
+import { searchModel, deleteModel, getAllModels } from '../../lib/api-client';
 import { Dispatcher } from 'consus-core/flux';
 
 export default class ModelController {
     static getModel(address) {
         return searchModel(address).then( model => {
-            Dispatcher.handleAction("MODEL_FOUND", model);
+            Dispatcher.handleAction('MODEL_FOUND', model);
         }).catch( () => {
-            Dispatcher.handleAction("NO_MODEL_FOUND");
+            Dispatcher.handleAction('NO_MODEL_FOUND');
+        });
+    }
+
+    static deleteModel(address){
+        return deleteModel(address).then(data => {
+            Dispatcher.handleAction('MODEL_DELETED', data.deletedModel);
+        }).catch(error => {
+            Dispatcher.handleAction('ERROR', {
+                error: error.message
+            });
+        }).then(() => {
+            return getAllModels().then(models => {
+                Dispatcher.handleAction("MODELS_RECEIVED", models);
```

Comment:
The items of model are deleted when a model is deleted. Items don't need to be updated since that would occur when the user goes to do something like view all items. The trigger here is so that the view all models page can update properly since that's where they trigger this action.